app/controllers/concerns/authentication.rb

```diff
@@ -59,8 +57,8 @@ def rbac_allowed?
     return true if ActiveModel::Type::Boolean.new.cast(Settings.disable_rbac)
     return valid_cert_auth? if identity_header.cert_based?
 
-    @rbac_api ||= ::RbacApi.new(raw_identity_header)
-    @rbac_api.check_user
+    set_authenticated_user
```

Comment:
Do we really need this method? Can't be hidden lazily in the next line? Also, where it is defined? :thinking:

Reply:
Without using the method, the `User.current` does not get initiated and so `current_user` is `nil`.

Defined right below:

```ruby
 66   def set_authenticated_user
 67     User.current = user
 68     Insights::API::Common::AuditLog.audit_with_account(
 69       current_user.account_number
 70     )
 71   end
```

Reply:
Okay, Ignore my comment here :see_no_evil: